collect_app/src/androidTest/java/org/odk/collect/android/feature/smoke/BadServerTest.java

```diff
@@ -42,4 +46,21 @@ public void whenHashNotIncludedInFormList_showError() {
                 )
                 .navigateBack();
     }
+
+    @Test
+    /*
+     A server that doesn't return hashes correctly for media files would fool Collect into thinking
+     there was a new one each time. This happens because we would compare the existing file's
+     (computed) hash with the server one assuming there is a prefix and end up comparing against
+     a substring.
+    */
+    public void whenMediaFileHasMissingPrefix_showsAsUpdated() {
```

Comment:
This is probably not the behaviour we want, but it is the behaviour we have. I wrote this test against `master` to double-check. I'll pull out another issue, so we can discuss how to improve this.

Reply:
https://github.com/getodk/collect/issues/4919